Tests/SourceKitLSPTests/InlayHintsTests.swift

```diff
@@ -0,0 +1,159 @@
+      trace: .off,
+      workspaceFolders: nil
+    ))
+  }
+
+  func performInlayHintsRequest(text: String, range: Range<Position>? = nil) -> [InlayHint] {
+    let url = URL(fileURLWithPath: "/\(#function)/a.swift")
+    
+    sk.send(DidOpenTextDocumentNotification(textDocument: TextDocumentItem(
+      uri: DocumentURI(url),
+      language: .swift,
+      version: 17,
+      text: text
+    )))
+
+    let request = InlayHintsRequest(textDocument: TextDocumentIdentifier(url), range: range)
+    return try! sk.sendSync(request)
+  }
+
+  func testEmpty() {
+    let text = ""
+    let hints = performInlayHintsRequest(text: text)
+    XCTAssertEqual(hints, [])
+  }
+
+  func testBindings() {
+    let text = """
+    let x = 4
+    var y = "test" + "123"
+    """
+    let hints = performInlayHintsRequest(text: text)
+    XCTAssertEqual(hints, [
+      InlayHint(
+        position: Position(line: 0, utf16index: 9),
+        category: .type,
+        label: "Int"
+      ),
+      InlayHint(
+        position: Position(line: 1, utf16index: 22),
+        category: .type,
+        label: "String"
+      ),
+    ])
+  }
+
+  func testRanged() {
+    let text = """
+    func square(_ x: Double) -> Double {
+      let result = x * x
+      return result
+    }
+
+    func collatz(_ n: Int) -> Int {
+      let even = n % 2 == 0
+      let result = even ? (n / 2) : (3 * n + 1)
+      return result
+    }
+    """
+    let range = Position(line: 6, utf16index: 0)..<Position(line: 9, utf16index: 0)
+    let hints = performInlayHintsRequest(text: text, range: range)
+    XCTAssertEqual(hints, [
+      InlayHint(
+        position: Position(line: 6, utf16index: 23),
+        category: .type,
+        label: "Bool"
+      ),
+      InlayHint(
+        position: Position(line: 7, utf16index: 43),
+        category: .type,
+        label: "Int"
+      )
+    ])
+  }
+
+  func testFields() {
+    let text = """
+    class X {
+      let instanceMember = 3
+      static let staticMember = "abc"
+    }
+
+    struct Y {
+      var instanceMember = "def" + "ghi"
+      static let staticMember = 1 + 2
+    }
+
+    enum Z {
+      static let staticMember = 3.0
+    }
+    """
+    let hints = performInlayHintsRequest(text: text)
+    XCTAssertEqual(hints, [
+      InlayHint(
+        position: Position(line: 1, utf16index: 24),
+        category: .type,
+        label: "Int"
+      ),
+      InlayHint(
+        position: Position(line: 2, utf16index: 33),
+        category: .type,
+        label: "String"
+      ),
+      InlayHint(
+        position: Position(line: 6, utf16index: 36),
+        category: .type,
+        label: "String"
+      ),
+      InlayHint(
+        position: Position(line: 7, utf16index: 33),
+        category: .type,
+        label: "Int"
+      ),
+      InlayHint(
+        position: Position(line: 11, utf16index: 31),
+        category: .type,
+        label: "Double"
+      ),
+    ])
+  }
```

Comment:
Could you also add a test case with variables that already have type annotations and add a `FIXME` to it that we shouldn’t be displaying the type hint there?